### PR TITLE
Implement expectNext() and test cases

### LINKED ISSUE #43: [parser] implement expectNext w/ test cases
```go
// ExpectNext helps to check whether next token is
// type of token we want, and if true then return it
// otherwise return with error
// TODO: implement me w/ test cases :-)
func expectNext(buf TokenBuffer, t TokenType) (bool, Token, error) {
	return false, Token{}, nil
}
```

### PR DESCRIPTION
**Pull request by boohyunsik**
resolved: #43

details: Implement expectNext() and test cases
And I have one question, how we can compare error?

- [x] Test case

### CODE REVIEW

**Comment by zeroFruit** [comment]
first, sorry... plz remove `Token` from returning statement. I made a mistake :(

**Comment by zeroFruit** [comment]
And if the code change is not about adding feature or modifying it. plz `git commit --amend` or squash the commit with force push
> Fix wrong typo in parser_internal_test.go

**Comment by boohyunsik** [comment]
> And if the code change is not about adding feature or modifying it. plz `git commit --amend` or squash the commit with force push
> 
> > Fix wrong typo in parser_internal_test.go

Ok I made it on purpose. I will do that way.

File: parse/parser.go [hunk 1]
```diff
@@ -91,9 +95,13 @@ func nextTokenIs(buf TokenBuffer, t TokenType) bool {
...
-// TODO: implement me w/ test cases :-)
-func expectNext(buf TokenBuffer, t TokenType) (bool, Token, error) {
-	return false, Token{}, nil
+func expectNext(buf TokenBuffer, t TokenType) (bool, error) {
+	ret := buf.Peek(NEXT).Type == t
+	var err error = nil
+	if !ret {
+		err = errors.New("expectNext() : expecting token and next token are different")
+	}
+	return ret, err
```
Comment (junbeomlee) [inline parse/parser.go]: how about 
```
	if !(buf.Peek(NEXT).Type == t) {
		return false, errors.New("expectNext() : expecting token and next token are different")
	}
	return true, nil
```

Comment (boohyunsik) [reply parse/parser.go]: Yeap.

**Review by junbeomlee — APPROVED** [review]
LGTM

**Pull request merged**